src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private int treshold;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        treshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        size = 0;
+        table = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+    }
+
+    private class Node<K,V> {
+        private int hashCode;
+        private K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(K key, V value, Node<K,V> next) {
+            this.hashCode = (key != null) ? key.hashCode() : 0;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > treshold) {
+            resize();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int index = indexFor(key);
+        Node<K,V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(key, value, null);
```

Comment:
```suggestion
            table[indexFor(key)] = new Node<>(key, value, null);
```
you can use indexFor() method and because of that your method putForNullKey() - redundant

Reply:
done